rawhttp-core/src/main/java/rawhttp/core/RawHttp.java

```diff
@@ -448,4 +448,17 @@ private RequestLine verifyHost(RequestLine requestLine, RawHttpHeaders.Builder h
         }
     }
 
+    private static String hostHeaderValueFor(URI uri) {
+        if (hasDefaultPort(uri)) {
+            return uri.getHost();
+        } else {
+            return uri.getHost() + ":" + uri.getPort();
+        }
+    }
+
+    private static boolean hasDefaultPort(URI uri) {
+        return (uri.getPort() < 0)
+                || ((uri.getPort() == 80) && ("http".equalsIgnoreCase(uri.getScheme())))
+                || ((uri.getPort() == 443) && ("https".equalsIgnoreCase(uri.getScheme())));
```

Comment:
Not a big deal, but `scheme` is defined as always being lower-case... so using `equals` would suffice... I will merge as is though as this is fine too.